decidim/decidim
# Ignore social share button links in external domain warning

#### :tophat: What? Why?

This PR disables the external domain warning in social share links. 

The reasoning is the following (from a @ahukkanen [comment](https://github.com/decidim/decidim/pull/9852#discussion_r1032202309)):

> I think maybe we should remove the external link warning from the share buttons completely?
>
> I don't think it makes much sense with those. The external link warning is generally useful e.g. for comments or proposals when the content is controlled by the participants, to prevent other users following malicious links.

While I was developing #9545 I thought about this and then forgot to implement that. This PR fixes that. 

I've changed this file so its consistent with the [new ExternalLink class](https://github.com/decidim/decidim/tree/develop/decidim-core/app/packs/src/decidim/redesigned_external_link.js), so we call both of these sister features in the same way (see [comment](https://github.com/decidim/decidim/pull/10121#discussion_r1034885798)). 

In the process I also migrated it from jQuery to plain Javascript. 

#### :pushpin: Related Issues
 
- Related to #9545 

#### Testing

1. Go to a page with social share buttons
2. Click on a button
3. See that you don't have the external domain warning anymore


:hearts: Thank you!
